motion_control/common_controller_ng/include/meta/MetaController.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "meta/BaseMetaController.hpp"
+#include "controller/Controller.hpp"
+
+namespace cogip {
+
+namespace motion_control {
+
+/// @brief Meta controllers are controllers containers to execute them in chain.
+/// @tparam INPUT_SIZE      number of inputs
+/// @tparam OUTPUT_SIZE     number of outputs
+/// @tparam NB_CONTROLLERS  maximum number of controllers
+template <
+    size_t INPUT_SIZE,
+    size_t OUTPUT_SIZE,
+    size_t NB_CONTROLLERS
+>
+class MetaController :
+    public BaseMetaController,
+    public Controller<INPUT_SIZE, OUTPUT_SIZE>,
+    private etl::list<BaseController *, NB_CONTROLLERS> {
+public:
+    /// @brief Controller core method. Meta controller executes all sub-controllers in chain.
+    void execute() override {
+        if (this->empty()) {
+            std::cerr << "Error: no controller added." << std::endl;
+            return;
+        }
+        if ((! this->empty()) && this->back()->nb_outputs() != OUTPUT_SIZE) {
+            std::cerr << "Error: First controller must have the same number of inputs (" << this->back()->nb_inputs() << ") "
+                      << "as the last controller (" << this->back()->nb_inputs() << ")." << std::endl;
+            return;
+        }
+        if (nb_controllers_ != NB_CONTROLLERS)
```

Comment:
Not sure to understand this condition...

Reply:
debug oversight, done